userspace/libsinsp/test/events_param.ut.cpp

```diff
@@ -313,3 +313,48 @@ TEST_F(sinsp_with_test_input, event_category)
 	ASSERT_EQ(evt->get_category(), EC_OTHER);
 	ASSERT_EQ(get_field_as_string(evt, "evt.category"), "other");
 }
+
+/* Check that enum flags are correctly handled,
+ * even when a single enum value is matched by multiple flags.
+ */
+TEST_F(sinsp_with_test_input, enumparams)
+{
+	add_default_init_thread();
+
+	open_inspector();
+	sinsp_evt* evt = NULL;
+	sinsp_evt_param* param = NULL;
+
+	/* `PPME_SOCKET_SOCKET_E` is a simple event that uses a PT_ENUMFLAGS32 (param 1) */
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SOCKET_SOCKET_E, 3, PPM_AF_UNIX, 0, 0);
+
+	param = evt->get_param(0);
+	ASSERT_EQ(*(uint32_t *)param->m_val, PPM_AF_UNIX);
+
+	const char *val_str = NULL;
+	evt->get_param_as_str(0, &val_str);
+	// Since the enum value "1" matches multiple flags values,
+	// we expect a space-separated list of them
+	ASSERT_STREQ(val_str, "AF_LOCAL AF_UNIX");
+}
+
+/* Check that bitmask flags are correctly handled
+ */
+TEST_F(sinsp_with_test_input, bitmaskparams)
+{
+	add_default_init_thread();
+
+	open_inspector();
+	sinsp_evt* evt = NULL;
+	sinsp_evt_param* param = NULL;
+
+	/* `PPME_SYSCALL_OPENAT_E` is a simple event that uses a PT_FLAGS32 (param 3) */
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_OPENAT_E, 4, 0, "/tmp/foo", PPM_O_RDONLY|PPM_O_CLOEXEC, 0);
```

Comment:
I know it's not ideal but please add `(int64_t)` or `(uint64_t)` to all params that are 64 bit and specified as literals

Reply:
Damn you are right, i forgot about your fix :/ thanks for reminding me!

Reply:
Solved, thank you @LucaGuerra !